### PR TITLE
Add details to Asynchronous UpDownCounter API

### PR DESCRIPTION
**Pull request by reyang**
Follow up #1617.

## Changes

Added details to the Asynchronous Gauge instrument. In #1617 these were left as TODO.

### CODE REVIEW

File: specification/metrics/api.md [hunk 1]
```diff
@@ -716,7 +716,111 @@ Example uses for Asynchronous UpDownCounter:
...
+[Measurement](#measurement)s. It will only be called when the Meter is being
+observed. Individual language client SHOULD define whether this callback
+function needs to be reentrant safe / thread safe or not.
+
+Note: Unlike [UpDownCounter.Add()](#add) which takes the increment/delta value,
+the callback function reports the absolute value of the Asynchronous
+UpDownCounter. To determine the reported rate the Asynchronous UpDownCounter is
+changing, the difference between successive measurements is used.
+
+The callback function SHOULD NOT take indefinite amount of time. If multiple
```
Comment (jsuereth) [inline specification/metrics/api.md:753]: similar question on # of exporters per-sdk

**Review by jsuereth — APPROVED** [review]

File: specification/metrics/api.md (hunk 1 above)
Comment (reyang) [reply specification/metrics/api.md:753]: Tracked by the discussion here https://github.com/open-telemetry/opentelemetry-specification/pull/1703#discussion_r632982981.

**Review by jmacd — APPROVED** [review]

**Review by weyert — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

File: specification/metrics/api.md (hunk 1 above)
Comment (SergeyKanzhelev) [reply specification/metrics/api.md:753]: I think the collection cycle suggestion from the thread is great and making this note would be beneficial for the spec. Is there work item to track this - I don't think we need to block this PR.

**Review by SergeyKanzhelev — APPROVED** [review]

**Comment by SergeyKanzhelev** [comment]
enough approvals, cool off period passed, merging. @reyang would you please address the comment about the number of calls to the callback per collection comment?

**Pull request merged**